integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/rdd/CarbonSIRebuildRDD.scala

```diff
@@ -321,6 +324,26 @@ class CarbonSIRebuildRDD[K, V](
           LOGGER.info("Closing compaction processor instance to clean up loading resources")
           processor.close()
         }
+
+        // delete all the old data files which are used for merging
+        splits.asScala.foreach { split =>
+          val carbonFile = FileFactory.getCarbonFile(split.getFilePath)
+          carbonFile.delete()
+        }
+
+        // delete the indexfile/merge index carbonFile of old data files
+        val segmentPath = FileFactory.getCarbonFile(indexTable.getSegmentPath(segmentId))
+        val indexFiles = segmentPath.listFiles(new CarbonFileFilter {
```

Comment:
Is list files needed again ? It is costly in S3a/OBS. It is better to keep the list of data files and index files participated in rebuild in memory and clean up after rebuild without calling list files again.

Reply:
actually, we will be having all the data files in splits, but we dont use index files for rebuilding, so i will check again if there any possibility to avoid this

Reply:
we will have the list of data files in the task but not the index files, i will try and fix in anothr PR